runtime/cmd/cmd.go

```diff
@@ -106,19 +108,15 @@ func PrepareChecker(
 					}
 				}
 
-				importChecker, err := checker.EnsureLoaded(
-					importedLocation,
-					func() *ast.Program {
-						imported, _ := PrepareProgramFromFile(stringLocation, codes)
-						return imported
-					},
-				)
-				if err != nil {
-					return nil, err
+				importedChecker, ok := checkers[importedLocation.ID()]
+				if !ok {
+					importedProgram, _ := PrepareProgramFromFile(stringLocation, codes)
+					importedChecker, _ = PrepareChecker(importedProgram, importedLocation, codes, must)
```

Comment:
should we return errors here ?

Reply:
Short answer: No, the functions panic on their own, the second return value is not an error.

These two functions, and some others, are written in an unidiomatic-Go style (not by me): The second return value is not an error – these functions already panic on their own if they fail, the second return value is a "must" function that can be used to panic in a way that has access to the code of all programs (`codes`), so that a pretty error can be printed to the terminal